### DISCUSSION TITLE
fontWeight type error

### DISCUSSION THREAD

**Discussion by sillyleo** [macaron-css/macaron#27] [General]
Hi, thanks for the library! I came from vanilla-extract to macaron for the friendlier setup and variant API.

However, I encountered an issue with typing:

<img width="220" alt="image" src="https://user-images.githubusercontent.com/105896/212805756-a98471b8-3de9-4653-958f-6b2f88b8921c.png">

<img width="505" alt="image" src="https://user-images.githubusercontent.com/105896/212805849-0cc88111-0886-4d25-8c2a-4fc81393a471.png">

Do you have any idea why I can't assign numbers to the fontWeights? Thanks

**Comment by sillyleo** [comment]
BTW is there docs for styleVariants?

**Reply by p7gg** [reply]
I don't think so. But the styleVariants should work exactly like `@vanilla-extract/css`, so I guess their docs should help you:
https://vanilla-extract.style/documentation/api/style-variants/

**Comment by p7gg** [comment]
I mean, it doesn't work cuz that is not what the API is expecting, it expects something like this:
```ts
type Tokens = {
    [key: string]: string | Tokens;
};
```

You should just pass the values as strings and it'll work.
```ts
fontWeight: {
    normal: '400'
}
```

**Reply by sillyleo** [reply]
Thanks! Exactly what I did later.

I found it's by design like this: https://github.com/vanilla-extract-css/vanilla-extract/discussions/314 (possibly related?)

**Reply by p7gg** [reply]
Totally

**Comment by Mokshit06** [comment]
Hey @sillyleo, font weight should be a string as recommended by @p7gg

**Comment by sillyleo** [comment]
Hi @Mokshit06 thanks again. Would like to hear your suggestion on the best way to type a component.


## Issue

https://github.com/sillyleo/nomimono/blob/main/src/Button/index.tsx

In my setup here, the vite plugin I used to generate/infer type doesn't work. ([config](https://github.com/sillyleo/nomimono/blob/main/vite.config.ts))

<img width="523" alt="image" src="https://user-images.githubusercontent.com/105896/213061758-67dd7fd4-3903-4506-a5a4-1234ebd04599.png">

The type does work in local env (I can get auto-completion and checking). However, the published package doesn't have that.

Is there a recommended setup? For example, import some type definition and extend the type with my own props like Stitches?

Thank!

**Reply by Mokshit06** [reply]
Hey, thanks for reporting this! macaron doesn't publish types for generated component, though I should add that. It'll get fixed today

**Reply by sillyleo** [reply]
Oh yes!!

**Reply by Mokshit06** [reply]
So I looked into this issue and it looks like it will require VE to update some of their internal types. If it gets fixed within a few days then I'll upstream it to macaron, or just reimplement that type for now
